### PR TITLE
Fix redefine "UErrorCode" error

### PR DESCRIPTION
**Pull request by ArturAralin**

### CODE REVIEW

File: icu-date.lua [hunk 1]
```diff
@@ -1,9 +1,12 @@
 local ffi = require("ffi")
 local detect_icu_version_suffix = require("icu-date.detect_icu_version_suffix")
 local icu_ffi_cdef = require("icu-date.ffi_cdef")
+local box = require("local")
```
Comment (Totktonada) [inline icu-date.lua]: What is this?

File: icu-date.lua [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 
 local icu_version_suffix = detect_icu_version_suffix()
-icu_ffi_cdef(ffi, icu_version_suffix)
+box.once('icu-date:init', function()
```
Comment (Totktonada) [inline icu-date.lua]: So at the second run you will not have needed ffi.C.* functions.

File: icu-date.lua (hunk 1 above)
Comment (ArturAralin) [reply icu-date.lua]: oops. Fix this

File: icu-date.lua (hunk 2 above)
Comment (ArturAralin) [reply icu-date.lua]: Hm.. Then how to make a package reloading?

`package.reload(x)`

Comment (Totktonada) [reply icu-date.lua]: @knazarov Can you suggest something here?

Comment (ArturAralin) [reply icu-date.lua]: @Totktonada what about `tcall`?

https://www.lua.org/pil/8.4.html

Comment (Totktonada) [reply icu-date.lua]: Whether you just want to check whether these functions are defined?

```
ffi = require('ffi')
ffi.C.kill
---
- error: '[string "return ffi.C.kill"]:1: missing declaration for symbol ''kill'''
...
ffi.cdef([[ void kill(int, int); ]])
ffi.C.kill
---
- 'cdata<void ()>: 0x7fce7202a890'
...
```

So, yep, pcall() does the trick. Choose one of defined functions and try to acquire it under pcall().

Comment (ArturAralin) [reply icu-date.lua]: I want to escape repeated call of `cdef`.
Tomorrow I'll try pcall

Comment (ArturAralin) [reply icu-date.lua]: @Totktonada check new implementation

Comment (ArturAralin) [reply icu-date.lua]: Re:
@Totktonada check new implementation

p.s. my dev server waiting for you)

File: icu-date.lua [hunk 3]
```diff
@@ -1,6 +1,7 @@
 local ffi = require("ffi")
 local detect_icu_version_suffix = require("icu-date.detect_icu_version_suffix")
 local icu_ffi_cdef = require("icu-date.ffi_cdef")
+local box = require("box")
```
Comment (Totktonada) [inline icu-date.lua]: Why?

File: icu-date/ffi_cdef.lua [hunk 4]
```diff
@@ -1,256 +1,259 @@
+
 return function(ffi, icu_version_suffix)
-  ffi.cdef([[
-  void* malloc(size_t size);
-  void free(void* ptr);
+  pcall(function ()
```
Comment (Totktonada) [inline icu-date/ffi_cdef.lua]: You need to pcall of usage of defined/undefined function. Check your code youself, it does not work:

```
tarantool> ffi = require('ffi')
---
...

tarantool> pcall(ffi.cdef, 'int kill (int, int);')
---
- true
...

tarantool> pcall(ffi.cdef, 'int kill (int, int);')
---
- true
...
```

Also you often don't need to create an extra capture, say pcall(function() foo(x, y, z) end) can be written as pcall(foo, x, y, z).

File: icu-date.lua (hunk 3 above)
Comment (ArturAralin) [reply icu-date.lua]: Just need clean this 👀

File: icu-date/ffi_cdef.lua (hunk 4 above)
Comment (ArturAralin) [reply icu-date/ffi_cdef.lua]: @Totktonada 

I had checked my code again. Look on this. When I trying to redefine `enum` `ffi.cdef` throws error

```
Tarantool 1.10.2-1-ge0017ad
type 'help' for interactive help
tarantool> ffi = require 'ffi'
---
...

tarantool> ffi.cdef('typedef enum UErrorCode { U_USING_FALLBACK_WARNING = -128 }')
---
...

tarantool> ffi.cdef('typedef enum UErrorCode { U_USING_FALLBACK_WARNING = -128 }')
---
- error: attempt to redefine 'UErrorCode'
...
```

Comment (Totktonada) [reply icu-date/ffi_cdef.lua]: Yep, I don't take a type into consideration. Okay so.

**Comment by Totktonada** [comment]
Before the patch:

```
icu_date = require('icu-date')
package.loaded['icu-date'] = nil
icu_date = require('icu-date')
---
- error: '...cu-date-tmp/.rocks/share/tarantool/icu-date/ffi_cdef.lua:2: attempt to
    redefine ''UErrorCode'' at line 8'
```

After:

```
icu_date = require('icu-date')
package.loaded['icu-date'] = nil
icu_date = require('icu-date')
---
...
```

So it works.

**Comment by Totktonada** [comment]
Thanks! Merged.

**Pull request merged**